### ISSUE TITLE
GGMAX - Editor Enhancement - (Behavior Markers)

### ISSUE THREAD

**Issue by Necrym59** [TheGameCreators/GameGuruRepo#4872]
When making maps due to the extended use and range of behaviors in max its hard to find one if your not sure where you attached it too.  Is it possible to have a "Behavior" marker in the editor just to show which objects have a behaviour attached to them. In the image sample the icons could possibly be generic marker or even an actual behavior image icon.
![script_markers](https://github.com/TheGameCreators/GameGuruRepo/assets/95015657/68b67029-d9d5-4c3e-b3ea-780045399444)

**Comment by davetgc** [comment]
Does anyone else have any thoughts on this?

**Comment by MonkeyFrogStudio** [comment]
I think it's a great idea, actually. It could be something that could be turned on/off (visually) in the view options panel. That way the editor viewport isn't constantly crowded with these markers.

**Comment by synchromesh62** [comment]
Y i think its a great idea also but agree with MF and On Off would also be perfect.

**Comment by AmenMoses** [comment]
Additionally when an entity is selected and it has blobby logic connections it would be helpful to see those connections or to at least highlight the other entities involved in those connections.

So for example if you selected a door which has a key and a switch connected have those entities highlighted or the connections show up even if not in 'blobby logic' mode (or whatever it is called).

**Comment by synchromesh62** [comment]
> Additionally when an entity is selected and it has blobby logic connections it would be helpful to see those connections or to at least highlight the other entities involved in those connections.
> 
> So for example if you selected a door which has a key and a switch connected have those entities highlighted or the connections show up even if not in 'blobby logic' mode (or whatever it is called).

Ye thats a good point .. When you select **Visual Logic Connections** you often see a room of spaghetti :)
Not sure its possible but along with Necryms Nodes to see the connection for the one clicked on rather than all would be awesome.

**Comment by MonkeyFrogStudio** [comment]
I was thinking something similar. What if the object with the behavior got a marker of a specific color, such as red (like Necrym shows) ... red indicating it is a PARENT behavior object, and any object CONNECTED to it via the visual logic connectors would get a similar marker, but in another color, such as green or blue, showing it is a CHILD object, connected by the visual logic system. So, you turn on the visibility of Behavior Markers in the View Options Panel (see my suggestion above) and you see all the PARENT or red markers for each object that has a behavior. Then, if you select one of them, if they have any CHILDREN, those suddenly get a blue or green (or whatever color) marker. 

As an example, let's say there's a door with a key connected to it via the visual logic system. Of course, that can't be seen in the standard editor (without turning on visual logic via the button). But you activate the Behavior Marker visibility in View Options and the door object suddenly has the RED behavior marker over it. If you then select the door, then the key suddenly gets a green behavior marker telling you the key object is a child of the door object (parent).

**Comment by fearlesswee** [comment]
This is an AWESOME idea. To prevent screen clutter maybe these behavior icons could also be distance-culled; so if you zoom out to a birds-eye view of the map, you won't see dozens of these little icons everywhere.

**Comment by mav3r1ck198-1** [comment]
hell of a useful idea, and i really like @MonkeyFrogStudio development of the concept

**Comment by Necrym59** [comment]
The idea was that when an icon is clicked it would only then show the actual links - keeping the complexity/clutter down. No need for color coding as only the instigator (parent) would have the icon and when clicked would show the links to children in that group, etc. No extra buttons needed anywhere - lets keep things simple and not over complicate things.

**Comment by OldFlak** [comment]
Yes, we need a better way to manage connections - showing every connection when clicking the connections icon is ok for a couple of connections - but real messy when you have complex things going on.

These scenes has a lot more to be added yet but is already like a can of spaghetti:
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/12246318/f297704b-04f7-4839-bf4b-d007a1735bfb)

![image](https://github.com/TheGameCreators/GameGuruRepo/assets/12246318/35ee5811-8c08-486a-9253-fb39d2bab040)

Blue-Prints idea is not that difficult to use really, but Necyms idea would be far better for MAX
